### DISCUSSION TITLE
About the size of picker

### DISCUSSION THREAD

**Discussion by Zaplae** [helix-editor/helix#3531] [Q&A]
Why the size of picker is not fullscreen? It could show more characters.
Compared with fullscreen picker, is the version now better?

**Comment by David-Else** [comment]
I noticed this, `ctrl-t` can toggle off the preview to enable you to read all the file names. 

Ideally the size could be adjusted by the user to taste, or maybe if it were just *slightly bigger* it would be perfect? It looks nice as it is, I am sure people with have different opinions.

**Comment by salman-farooq-sh** [comment]
I think the picker should cover the whole screen given that it needs to display so much information. Another thing can be to change the split ratio between the list on the left and the preview on the right but that could be more difficult to do I suppose. Fullscreen picker would be great.

**Comment by Axlefublr** [comment]
I'm currently trying to change the default / allow for configuration in my [fork](https://github.com/Axlefublr/helix), for this exact question. \
However rust-analyzer is being dumb, so I can't find the place where the `area` parameter to the `render` method of `Picker`, defined in `helix-term/src/ui/picker.rs`, is passed. \
If someone could help me figure this out, I'd massively appreciate it! Will mention you in the readme of the fork :3

**Reply by satoqz** [reply]
I had a short look, pickers are wrapped in an overlay that adds the surrounding margins (see [`overlaid`](https://github.com/helix-editor/helix/blob/be2884d80061493ab1999d73ffa51dfb4f59639d/helix-term/src/ui/overlay.rs#L18-L24)). The wrapped pickers are then simply [pushed as a full compositor layer](https://github.com/helix-editor/helix/blob/be2884d80061493ab1999d73ffa51dfb4f59639d/helix-term/src/commands/lsp.rs#L419), which means that their `render` chain receives an `area` that is essentially the entire screen (The `area` is passed [here](https://github.com/helix-editor/helix/blob/be2884d80061493ab1999d73ffa51dfb4f59639d/helix-term/src/compositor.rs#L176-L180)).

A quick hack to make all pickers full screen would be:

```diff
pub fn overlaid<T>(content: T) -> Overlay<T> {
    Overlay {
        content,
-       calc_child_size: Box::new(|rect: Rect| clip_rect_relative(rect.clip_bottom(2), 90, 90)),
+       calc_child_size: Box::new(|rect: Rect| clip_rect_relative(rect.clip_bottom(0), 100, 100)),
    }
}
```

So far `overlaid` is only used for pickers, so you're not breaking any other component (yet) (I think).

Hope this addresses your question, this is my first time digging into the Helix codebase.

**Reply by Axlefublr** [reply]
omg thank you so much! such a detailed response, too!
I'll follow this information and see if I can make picker fullscreening configurable, in a reasonable way
Regardless, I will indeed mention you in the readme! Once again thank you :3

**Comment by salman-farooq-sh** [comment]
Related: https://github.com/helix-editor/helix/discussions/11780

Sorry forgot to mention it here.

**Comment by Axlefublr** [comment]
For everyone who wants your pickers to be fullscreen: try my [fork](https://github.com/Axlefublr/helix)! \
This was possible thanks to @satoqz's help in the above thread! 🥳
